pkg/operators/process/process.go

```diff
@@ -0,0 +1,777 @@
+				}
+			}
+		}
+	}
+
+	return procInfo, nil
+}
+
+// unescapeCommand unescapes the command string according to kernel escaping rules
+// %ESCAPE_SPACE: ('\f', '\n', '\r', '\t', '\v')
+// %ESCAPE_SPECIAL: ('\"', '\\', '\a', '\e')
+func unescapeCommand(cmd string) string {
+	// If there are no escape sequences, return the original string
+	if !strings.Contains(cmd, "\\") {
+		return cmd
+	}
+
+	var builder strings.Builder
+	builder.Grow(len(cmd)) // Pre-allocate capacity to avoid reallocations
+
+	for i := 0; i < len(cmd); i++ {
+		// Check for escape sequence
+		if cmd[i] == '\\' && i+1 < len(cmd) {
+			switch cmd[i+1] {
+			// Special whitespace characters
+			case 'f':
+				builder.WriteByte('\f')
+			case 'n':
+				builder.WriteByte('\n')
+			case 'r':
+				builder.WriteByte('\r')
+			case 't':
+				builder.WriteByte('\t')
+			case 'v':
+				builder.WriteByte('\v')
+			// Special characters
+			case '"':
+				builder.WriteByte('"')
+			case '\\':
+				builder.WriteByte('\\')
+			case 'a':
+				builder.WriteByte('\a')
+			case 'e':
+				builder.WriteByte(27) // ASCII 27 for escape character
+			default:
+				// Unknown escape sequence, keep both characters
+				builder.WriteByte('\\')
+				builder.WriteByte(cmd[i+1])
+			}
+			i++ // Skip the next character as we've already processed it
+		} else {
+			builder.WriteByte(cmd[i])
+		}
+	}
+
+	return builder.String()
+}
+
+// parseStatFile parses the /proc/[pid]/stat file, handling the comm field correctly
+// The comm field is enclosed in parentheses and may contain spaces
+func parseStatFile(statData []byte, result [][]byte) error {
+	// The format of /proc/[pid]/stat is:
+	// pid (comm) state ppid ... and more fields
+	// The challenge is that comm can contain spaces and parentheses
+
+	// First, find the closing parenthesis for the comm field
+	commEndIdx := bytes.LastIndex(statData, []byte{')'})
+	if commEndIdx == -1 {
+		return fmt.Errorf("invalid stat file format: missing closing parenthesis")
+	}
+
+	// Find the opening parenthesis
+	commStartIdx := bytes.Index(statData, []byte{'('})
+	if commStartIdx == -1 {
+		return fmt.Errorf("invalid stat file format: missing opening parenthesis")
+	}
+
+	// Validate that the opening parenthesis comes before the closing one
+	if commStartIdx >= commEndIdx {
+		return fmt.Errorf("invalid stat file format: parentheses in wrong order")
+	}
+
+	// Extract the pid (field 1)
+	pidStr := bytes.TrimSpace(statData[:commStartIdx])
+
+	// Extract the comm (field 2)
+	comm := statData[commStartIdx+1 : commEndIdx]
+
+	// Extract the rest of the fields
+	// We only need a few specific fields, so we'll parse them directly
+	// without creating a full array of all fields
+	restData := statData[commEndIdx+1:]
+	fields := bytes.Fields(restData)
+
+	// We need at least 21 fields after the comm field to access field 22
+	if len(fields) < 21 {
+		return fmt.Errorf("invalid stat file format: not enough fields")
+	}
```

Comment:
Also, do we really need this check? Is it too bad to make it crash if the passed array isn't big enough?

Reply:
Yeah, I think it's bad to crash on a gadget level (which this context would run in) at some random point on runtime. Controlled crashes should IMHO only happen on startup or if we know the system is really in a bad state anyways (this could happen due to an update that changed the file layout, although I'm sure that this API is pretty stable).